ui-lib/src/components/ProgressBar/index.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { memo } from "react";
+import styled from "styled-components";
+import { border, BorderProps, color, ColorProps, space, SpaceProps } from "styled-system";
+import CheckAlone from "@ui/assets/icons/CheckAloneMedium";
+import Text from "@components/Text";
+import Flex from "@components/Layout/Flex";
+
+export interface Props {
+  /**
+   * An array of labels that will determine the progress bar steps.
+   */
+  steps: string[];
+  /**
+   * Index of the active step, starting at zero and defaulting to 0 if omitted.
+   */
+  activeIndex?: number;
+}
+/**
+ * The state of a progress bar step.
+ */
+type StepState = "pending" | "current" | "completed";
+export type StepProps = {
+  /**
+   * State of the step.
+   */
+  state: StepState;
+  /**
+   * The label to display.
+   */
+  label: string;
+  /**
+   * If true, hides the left "separator" bar that bridges the gap between the wider separator and the item.
+   */
+  hideLeftSeparator: boolean;
+  /**
+   * The next step state, or undefined if the current step is the last one.
+   */
+  nextState?: StepState;
+};
+
+export const Item = {
+  Container: styled.div.attrs(_ => ({
+    mx: 2,
+  }))<ColorProps & BorderProps & SpaceProps>`
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    width: 24px;
+    height: 24px;
+    ${color}
+    ${border}
+    ${space}
+  `,
+  Spacer: styled.div<SpaceProps>`
+    display: flex;
+    align-self: stretch;
+    ${space}
+  `,
+  Current: styled.div.attrs(_ => ({
+    backgroundColor: "palette.v2.primary.dark",
+  }))<ColorProps>`
+    width: 6px;
+    height: 6px;
+    border-radius: 9999px;
```

Comment:
🤔 ?

Reply:
Haha yeah it might look weird, it's a habit I took when I want to round something safely [after being bitten](https://github.com/tailwindlabs/tailwindcss/issues/1949) by `border-radius: 50%`. In this case it's overkill, I'll change to something more sensible 👍.